packages/hoc-input/src/formInput.js

```diff
@@ -0,0 +1,175 @@
+
+const formInput = WrappedComponent => {
+  return class Input extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        value: props.value,
+        focused: props.focused,
+        validationMessages: props.validationMessages
+      };
+
+      this.onChange = this.onChange.bind(this);
+      this.onInvalid = this.onInvalid.bind(this);
+      this.onFocus = this.onFocus.bind(this);
+      this.onBlur = this.onBlur.bind(this);
+      this._renderInput = this._renderInput.bind(this);
+      this._renderFailedMessages = this._renderFailedMessages.bind(this);
+    }
+
+    render() {
+      const {
+        label,
+        focused,
+        validationMessages,
+        errorIconSrc,
+        onChange,
+        onFocus,
+        onBlur,
+        onInvalid,
+        ...wrappedComponentProps
+      } = this.props;
+      const wrapperProps = {
+        focused: this.state.focused,
+        invalid: this.state.validationMessages.length > 0,
+        filled: this.state.value.length > 0
+      };
+      return (
+        <Wrapper {...wrapperProps}>
+          {this._renderInput(wrappedComponentProps)}
+          {this._renderLabel(label)}
+          {this._renderFailedMessages(wrapperProps.invalid)}
+        </Wrapper>
+      );
+    }
+
+    _renderInput(inputProps) {
+      const handlers = {
+        onChange: this.onChange,
+        onInvalid: this.onInvalid,
+        onFocus: this.onFocus,
+        onBlur: this.onBlur
+      };
+      return (
+        <div className="input">
+          <WrappedComponent
+            {...inputProps}
+            {...handlers}
+            value={this.state.value}
+          />
+        </div>
+      );
+    }
+
+    _renderLabel(label, focused) {
+      return (
+        label && (
+          <Text
+            title
+            className="label"
+            size={fontSizes.SMALL}
+            lineHeight={1.14}
+          >
+            {label}
+          </Text>
+        )
+      );
+    }
+
+    _renderFailedMessages(invalid) {
+      return (
+        invalid && (
+          <div className="messages">
+            {this.state.validationMessages.map((text, index) => (
+              <div className="message" key={`err-${index}`}>
+                <Image
+                  className="icon"
+                  src={this.props.errorIconSrc}
+                  badge
+                  size={badgeSizes.SMALL}
+                />
+                <Text
+                  lineHeight={`${badgeSizes.SMALL}px`}
+                  size={fontSizes.MEDIUM}
+                >
+                  {text}
+                </Text>
+              </div>
+            ))}
+          </div>
+        )
+      );
+    }
+    onChange(event) {
+      const validationMessages =
+        this.props.validationMessages.length > 0
+          ? this.props.validationMessages
+          : [];
+
+      this.setState({
+        value: event.target.value,
```

Comment:
Got this error using this HOC on a stripe component:

![image](https://user-images.githubusercontent.com/17936244/34176961-00d12038-e4e9-11e7-8e58-94d32014e6ea.png)

Stripes calls `onChange` passing this object:
![image](https://user-images.githubusercontent.com/17936244/34177026-3fc69f5c-e4e9-11e7-8623-aca60a45ac97.png)

See reference on https://stripe.com/docs/stripe-js/reference#element-on